addons/gamification/models/gamification_challenge.py

```diff
@@ -456,16 +404,16 @@ def _generate_goals_from_challenge(self, cr, uid, ids, context=None):
 
                 for user_id in user_without_goal_ids:
                     values.update({'user_id': user_id})
-                    goal_id = goal_obj.create(cr, uid, values, context=context)
-                    to_update.append(goal_id)
+                    goal = Goal.create(values)
+                    to_update.append(goal.id)
```

Comment:
Instead of creating a list, pool the gamification.goal from registry(env[gamification.goal]) and add goal into it and then call update_goal method